engine/processeventvisitor.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+namespace pong
+{
+
+class Button;
+class CheckBox;
+class Slider;
+class Text;
+
+class ProcessEventVisitor
+{
+public:
+    virtual void VisitButton(Button& button) = 0;
```

Comment:
It would be more expected to have all of these just called "Visit", the type of the param makes it obvious what they are visiting